### PR TITLE
documentation #801: updated swagger doc for Poll endpoints:

### LINKED ISSUE #801: Update Swagger doc for Poll (1)
We are updating the documentation for all endpoints and adding more _description and details_. 

One of the main controllers is Poll. You should update the documentation according to the **pattern** below:
```ts
  @ApiBearerAuth()
  @ApiOkResponse()
  @ApiBadRequestResponse({
    description: `\n
    InvalidEntityIdException:
      User with such id is not found

    AlreadyRegisteredException:
      User is already registered

    InvalidBodyException:
      Group id can not be empty
      IsCaptain must be a boolean
      IsCaptain can not be empty
    
    AbsenceOfCaptainException:
      Captain was not found
    
    AbsenceOfCaptainTelegramException:
      Captain's telegramId was not found`,
  })
  @ApiUnauthorizedResponse({
    description: `\n
    UnauthorizedException:
      Unauthorized`,
  })
  @ApiForbiddenResponse({
    description: `\n
    NoPermissionException:
      You do not have permission to perform this action
    
    ForbiddenException:
      Forbidden`,
  })
  @ApiParam({
    name: 'userId',
    required: true,
    description: 'Id of a user to request to the group',
  })
  @ApiEndpoint({
    summary: 'Request captain or admin to join the group',
    permissions: PERMISSION.USERS_$USERID_GROUP_REQUEST,
  })
  @Patch('/:userId/requestNewGroup')
  requestNewGroup (
    @Param('userId', UserByIdPipe) userId: string,
    @Body() body: GroupRequestDTO,
  ) {
    return this.userService.requestNewGroup(userId, body);
  }
```
**Structure:**
- auth (if needed)
- ok response
- client errors
- params/query description
- api endpoint for summary, permissions, guards

You should also add **field description** in the bady or response.
Add **summary** to endpoints (fix this issue).
Set **default values** for the optional fields

image

**Example**: #777

_My DM is always open for you. Happy coding_ 🐸

[Our Swagger](https://apidev.fictadvisor.com/api)
[Doc](https://docs.nestjs.com/openapi/introduction)

### PR DESCRIPTION
**Pull request by Dmutre**
- POST v2/poll/questions
- GET v2/poll/teachers/{userId}
- DELETE v2/poll/questions/{questionId}
- PATCH v2/poll/questions/{questionId}
- GET v2/poll/questions/{questionId}
- POST v2/poll/questions/{questionId}/roles
- DELETE v2/poll//questions/{questionId}/roles/{role}

closes #801

### CODE REVIEW

File: src/v2/api/controllers/PollController.ts [hunk 1]
```diff
@@ -119,38 +181,71 @@ export class PollController {
...
+    InvalidEntityIdException:
+      question with such id is not found`,
+  })
+  @ApiParam({
+    name: 'questionId', 
+    required: true,
+    description: 'Id of question you want to get',
   })
+  @ApiEndpoint({
+    summary: "Request to get a question by ID",
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/PollController.ts:198]: ESLint: Strings must use singlequote. (quotes)

File: src/v2/api/controllers/PollController.ts [hunk 2]
```diff
@@ -119,38 +181,71 @@ export class PollController {
...
-  @Get('/questions/:questionId')
   @ApiOkResponse({
     type: QuestionWithRolesResponse,
   })
   @ApiBadRequestResponse({
-    description: `InvalidEntityIdException:\n
-                  question with such id is not found`,
+    description: `\n
+    InvalidEntityIdException:
+      question with such id is not found`,
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/PollController.ts:190]: Upper letter (`Question`), not `question`

File: src/v2/api/controllers/PollController.ts [hunk 3]
```diff
@@ -119,38 +181,71 @@ export class PollController {
...
+      Visibility must be boolean
+
+    InvalidBodyException:
+      Visibility can not be empty
+
+    InvalidBodyException:
+      Requirement must be boolean
+
+    InvalidEntityIdException:
+      Question with such id is not found`,
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/PollController.ts:227]: * Don't split `InvalidBodyException` into separate parts. There should be only one `InvalidBodyException`  and under it you should write name of your exception, e.g `Visibility cannot be empty`
* Rename: `Role cannot be empty` `Visibility cannot be empty`

File: src/v2/api/controllers/PollController.ts [hunk 4]
```diff
@@ -119,38 +181,71 @@ export class PollController {
...
   })
   @ApiForbiddenResponse({
-    description: `NoPermissionException:\n
-                  You do not have permission to perform this action`,
+    description: `\n
+    NoPermissionException:
+      You do not have permission to perform this action`,
+  })
+  @ApiParam({
+    name: "questionId",
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/PollController.ts:240]: ESLint: Strings must use singlequote. (quotes)

File: src/v2/api/controllers/PollController.ts [hunk 5]
```diff
@@ -119,38 +181,71 @@ export class PollController {
...
+    required: true,
+    description: 'Id of question in wich you want to create role',
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/PollController.ts:242]: Rename: `Id of the question to which you want to attach the role`

File: src/v2/api/controllers/PollController.ts [hunk 6]
```diff
@@ -119,38 +181,71 @@ export class PollController {
...
+  })
+  @ApiEndpoint({
+    summary: 'Request to create role in question',
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/PollController.ts:245]: Rename: `Give the role to the question by id`

File: src/v2/api/controllers/PollController.ts [hunk 7]
```diff
@@ -159,9 +254,26 @@ export class PollController {
...
+  @ApiUnauthorizedResponse({
+    description: `\n
+    UnauthorizedException:
+      Unauthorized`,
+  })
+  @ApiForbiddenResponse({
+    description: `\n
+    NoPermissionException:
+      You do not have permission to perform this action`,
+  })
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/PollController.ts:276]: Add description to `questionId` and `role`

File: src/v2/api/controllers/PollController.ts [hunk 8]
```diff
@@ -172,23 +284,15 @@ export class PollController {
...
-    description: `InvalidEntityIdException:\n
-                  Question with such id is not found
-                  QuestionRole is not found`,
-  })
-  @ApiForbiddenResponse({
-    description: `NoPermissionException:\n
-                  You do not have permission to perform this action`,
+  @ApiEndpoint({
+    summary: 'Request to delete question',
+    permissions: PERMISSION.QUESTIONS_ROLES_DELETE,
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/PollController.ts:289]: Wrong summary. This endpoint does not delete the question

**Review by SharpDevOps10 — CHANGES_REQUESTED** [review]
* If you describe class that extends another => you should describe `extended` class as well.
* In some cases, you should change `InvalidEntityIdException` argument. The argument should begin with a capital letter (it concerns QuestionByRoleAndIdPipe, QuestionByIdPipe)

File: src/v2/api/dtos/CreateQuestionWithRolesDTO.ts [hunk 9]
```diff
@@ -9,6 +9,7 @@ export class CreateQuestionWithRolesDTO extends CreateQuestionDTO {
   @Type(() => CreateQuestionRoleDTO)
     @ApiProperty({
       type: [CreateQuestionRoleDTO],
```
Comment (fokaaas) [inline src/v2/api/dtos/CreateQuestionWithRolesDTO.ts:11]: You must also describe the class from which you extend (CreateQuestionDTO).

File: src/v2/api/responses/QuestionWithRolesResponse.ts [hunk 10]
```diff
@@ -3,11 +3,14 @@ import { CreateQuestionRoleDTO } from '../dtos/CreateQuestionRoleDTO';
 import { QuestionResponse } from './QuestionResponse';
 
 export class QuestionWithRolesResponse extends QuestionResponse {
```
Comment (fokaaas) [inline src/v2/api/responses/QuestionWithRolesResponse.ts:5]: You must also describe the class from which you extend (QuestionResponse).

File: src/v2/api/responses/PollDisciplineTeachersResponse.ts [hunk 11]
```diff
@@ -39,11 +39,14 @@ class DisciplineTeachers {
...
+  @ApiProperty({
+    description: 'Show was teacher selected in last semester',
+  })
     hasSelectedInLastSemester:boolean;
 
   @ApiProperty({
     type: [DisciplineTeachers],
+    description: 'Information about the teacher\'s discipline',
   })
     teachers: DisciplineTeachers[];
```
Comment (fokaaas) [inline src/v2/api/responses/PollDisciplineTeachersResponse.ts:51]: Also describe DisciplineReachers & Subject

File: src/v2/api/dtos/CreateQuestionRoleDTO.ts [hunk 12]
```diff
@@ -6,18 +6,23 @@ import { ApiProperty } from '@nestjs/swagger';
 export class CreateQuestionRoleDTO {
   @ApiProperty({
     enum: TeacherRole,
+    description: 'An enum of teacher\'s',
```
Comment (fokaaas) [inline src/v2/api/dtos/CreateQuestionRoleDTO.ts:9]: teacher roles)

**Review by fokaaas — CHANGES_REQUESTED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**